src/services/services.ts

```diff
@@ -640,8 +660,21 @@ class SymbolObject implements Symbol {
     contextualSetAccessorTags?: JSDocTagInfo[];
 
     constructor(flags: SymbolFlags, name: __String) {
+        // Note: must match Symbol in src/compiler/types.ts
         this.flags = flags;
         this.escapedName = name;
+        this.declarations = undefined;
+        this.valueDeclaration = undefined;
+        this.id = 0;
+        this.mergeId = 0;
+        this.parent = undefined;
+        this.members = undefined;
+        this.exports = undefined;
+        this.exportSymbol = undefined;
+        this.constEnumOnlyModule = undefined;
+        this.isReferenced = undefined;
+        this.lastAssignmentPos = undefined;
+        (this as any).links = undefined; // used by TransientSymbol

```

Comment:
Unlike the `Symbol` "constructor" in compiler, we can actually define `NodeLinks` on `SymbolObject` and remove the `as any`.

Reply:
Good point!